### PR TITLE
Filter incoming messages on product name

### PR DESCRIPTION
**Pull request by adybbroe**
This PR is introducing a requirement on the product name in the incoming messages.

 - [ ] Closes #xxxx 
 - [x] Tests added 
 - [ ] Tests passed: Passes ``pytest`` 
 - [x] Passes ``flake8`` 
 - [ ] Fully documented

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -10,10 +10,10 @@ jobs:
       fail-fast: true
       matrix:
         os: ["ubuntu-latest", "macos-latest"]
-        python-version: ["3.9", "3.10", "3.11"]
+        python-version: ["3.9", "3.11", "3.12"]
```
Comment (mraspaud) [inline .github/workflows/ci.yaml:13]: Staying on 3.9?

File: activefires_pp/spatiotemporal_alarm_filtering.py [hunk 2]
```diff
@@ -180,6 +181,9 @@ def run(self):
                 elif msg.type not in ['file', 'collection', 'dataset']:
                     LOG.debug("Message type not supported: %s", str(msg.type))
                     continue
+                elif product_list and msg.data.get('product') not in product_list:
```
Comment (mraspaud) [inline activefires_pp/spatiotemporal_alarm_filtering.py:184]: what if there is not product in the message? is None a possible value in the product_list?

File: activefires_pp/spatiotemporal_alarm_filtering.py [hunk 3]
```diff
@@ -500,6 +492,13 @@ def distance_and_time_from_geojson_position(position, filepath):
     return obstime, dist
 
 
+def get_distance_between_two_points(geo_point1, geo_point2):
```
Comment (mraspaud) [inline activefires_pp/spatiotemporal_alarm_filtering.py:495]: nice refactoring!

File: activefires_pp/tests/test_messaging.py [hunk 4]
```diff
@@ -61,7 +61,7 @@ def get_fake_publiser(portnumber=1979):
 #     mymask_file = "/my/shape/file/with/polygons/to/filter/out"
 
 #     afpp = ActiveFiresPostprocessing(myconfigfile, myborders_file, mymask_file)
-#     afpp.publisher = get_fake_publiser()
+#     afpp.publisher = get_fake_publisher()
```
Comment (mraspaud) [inline activefires_pp/tests/test_messaging.py:64]: is this commented code necessary?

**Review by mraspaud — COMMENTED** [review]
A few comments, otherwise looks good.

File: activefires_pp/spatiotemporal_alarm_filtering.py (hunk 2 above)
Comment (adybbroe) [reply activefires_pp/spatiotemporal_alarm_filtering.py:184]: Ah good point! `None` is theoretically a possible name for a product, and then if the message does not contain the product name any product would pass through and give rise to a notification.

Comment (adybbroe) [reply activefires_pp/spatiotemporal_alarm_filtering.py:184]: Now I have changed so no product name in the message will check for products named "unknown" against the configured product names. You think this is better and sufficient?

**Comment by adybbroe** [comment]
Better now @mraspaud ?

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (mraspaud) [reply .github/workflows/ci.yaml]: Yep, it's not needed anymore, see here: https://scientific-python.org/specs/spec-0000/#support-window

**Review by mraspaud — APPROVED** [review]
LGTM

**Pull request merged**